### PR TITLE
feat: expose `allow_overlapping` property

### PR DESCRIPTION
**Pull request by frascuchon**
# Description

This PR adds support for configure overlapping for span questions on creation and update.

Refs https://github.com/argilla-io/argilla/issues/4694

**Type of change**

(Please delete options that are not relevant. Remember to title the PR according to the type of change)

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactor (change restructuring the codebase without changing functionality)
- [X] Improvement (change adding some improvement to an existing functionality)

**How Has This Been Tested**

(Please describe the tests that you ran to verify your changes. And ideally, reference `tests`)

- [ ] Test A
- [ ] Test B

**Checklist**

- [ ] I added relevant documentation
- [ ] I followed the style guidelines of this project
- [ ] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I filled out [the contributor form](https://tally.so/r/n9XrxK) (see text above)
- [ ] I have added relevant notes to the `CHANGELOG.md` file (See https://keepachangelog.com/)

### CODE REVIEW

**Comment by frascuchon** [comment]
> @frascuchon we don't want to change the span question responses validator so it checks that response values are not overlapping when the question settings have the overlapping to `false`?

Yes. But I would like to tackle this in a separate PR. I've created also 2 separated tasks https://github.com/argilla-io/argilla/issues/1750

File: src/argilla_server/validators/questions.py [hunk 1]
```diff
@@ -77,6 +83,8 @@ def _validate_question_settings(self, question_settings: QuestionSettings):
         self._validate_question_settings_label_options(question_settings, self._question_update.settings)
         self._validate_question_settings_visible_options(question_settings, self._question_update.settings)
 
+        self._validate_span_question_settings(question_settings, self._question_update.settings)
```
Comment (jfcalvo) [inline src/argilla_server/validators/questions.py:86]:
```suggestion
        self._validate_span_question_settings(question_settings, self._question_update.settings)
```

File: tests/unit/api/v1/questions/test_update_question.py [hunk 2]
```diff
@@ -111,3 +111,32 @@ async def test_update_question_with_more_visible_options_than_allowed(
         assert response.json() == {
             "detail": "the value for 'visible_options' must be less or equal to the number of items in 'options' (3)"
         }
+
+    async def test_update_span_question_with_unsupported_allow_overlapping_value(
```
Comment (jfcalvo) [inline tests/unit/api/v1/questions/test_update_question.py:115]:
```suggestion
    async def test_update_span_question_disabling_allow_overlapping(
```

Comment (jfcalvo) [reply tests/unit/api/v1/questions/test_update_question.py:115]: Maybe another one checking that it's possible to enable `allow_overlapping`?

Something like `test_update_span_question_enabling_allow_overlapping`.

File: src/argilla_server/validators/questions.py [hunk 3]
```diff
@@ -127,6 +135,20 @@ def _validate_question_settings_visible_options(
...
+    def _validate_span_question_settings(
+        self, question_settings: SpanQuestionSettings, question_settings_update: QuestionSettingsUpdate
+    ) -> None:
+        if question_settings_update.type != QuestionType.span:
+            return
+
+        if (
+            question_settings.allow_overlapping is True
+            and question_settings_update.allow_overlapping != question_settings.allow_overlapping
+        ):
```
[the remark this answers is not in the document]
Comment (jfcalvo) [reply src/argilla_server/validators/questions.py:147]: Optionally if you want to move the logic to a private function to make it more readable:

```suggestion
        if self._is_disabling_allow_overlapping():
```

**Review by jfcalvo — APPROVED** [review]

**Pull request merged**